gcputil/gcputil.go

```diff
@@ -0,0 +1,120 @@
+// Package gcputil provides functions
+// that interact with various GCP platforms
+// such as getting GAE metadata and registering
+// stack driver monitoring
+package gcputil
```

Comment:
I'm a bit hesitant of the `util` name here as the contents seem to be mostly related to observability/monitoring via  Stackdriver. Names are hard, though, and `observability` _may_ be a controversial choice of package name (beyond being a lot to type).

Since this is all Stackdriver oriented, maybe we could go that route and call it `stackdriver`? I worry about this one as it may conflict with other packages of that name.

You have any thoughts or other ideas?

Reply:
If we're happy with `stackdriver.IsGAE()`, then sure. But functions like `IsGAE()` and `ProjectID()` are broad GCP functions and not StackDriver specific -- I try to avoid naming discussions so I'm happy with whatever you suggest lol

Reply:
I think Stackdriver is related to a GCP project, which I think is OK.

However, if we decide to hook the `prometheus` stuff I mentioned in your other issue, that name will definitely _not_ work 😓 

Perhaps we do something more generic like `monitoring` or `observability` but those seem like long names and `o11y` is probably going to get us a lot of pain on Twitter 😅 

Let's toss this one to the internal community.

Reply:
and it's not all related to GCP, since SD can have AWS EC2 resource

Reply:
@gaplyk I think `gcputil.NewOpenCensusExporter()` still makes sense from within an AWS application, since StackDriver *is* a GCP product...but I'm really happy either way, as long as I can use whichever package to instantiate an exporter.